library/block/block_content_registry/src/main/java/org/quiltmc/qsl/block/content/registry/mixin/EnchantingTableBlockMixin.java

```diff
@@ -18,22 +18,30 @@
 
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.Redirect;
 
-import net.minecraft.block.Block;
 import net.minecraft.block.BlockState;
 import net.minecraft.block.EnchantingTableBlock;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.random.RandomGenerator;
 import net.minecraft.world.World;
 
 import org.quiltmc.qsl.block.content.registry.api.BlockContentRegistries;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
 
 @Mixin(EnchantingTableBlock.class)
 public class EnchantingTableBlockMixin {
-	@Redirect(method = "isValidForBookshelf", at = @At(value = "INVOKE", target = "Lnet/minecraft/block/BlockState;isOf(Lnet/minecraft/block/Block;)Z"))
-	private static boolean hasEnchantmentPower(BlockState blockState, Block ignored) {
-		return BlockContentRegistries.ENCHANTING_BOOSTERS.get(blockState.getBlock()).orElse(0f) != 0f;
+	@Inject(method = "isValidForBookshelf", at = @At("HEAD"), cancellable = true)
+	private static void hasEnchantmentPower(World world, BlockPos pos, BlockPos offset, CallbackInfoReturnable<Boolean> cir) {
+		BlockPos blockPos = pos.add(offset);
+		BlockState state = world.getBlockState(blockPos);
+		float power = BlockContentRegistries.ENCHANTING_BOOSTERS.get(state.getBlock()).map(booster -> booster.getEnchantingBoost(world, state, blockPos)).orElse(0f);
+		boolean hasPower = power >= 0f && world.isAir(pos.add(offset.getX() / 2, offset.getY(), offset.getZ() / 2));
```

Comment:
Hm, maybe these 2 should be 2.0F? i do remember having trouble with dividing with integers before

Reply:
This is fine. We want integer division here anyway.